src/core/config/type-cast/type-casters/number.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * @prettier
+ */
+const numberTypeCaster = (value) => {
+  const parsedValue = parseInt(value)
```

Comment:
Let's define explicit `radix` to avoid various side effects:

```suggestion
  const parsedValue = parseInt(value, 10)
```